### ISSUE TITLE
(node:26) Warning: a promise was created in a handler at app/node_modules/bull/lib/queue.js:732:15 but was not returned from it, see http://goo.gl/rRqMUw     at new Promise (/app/node_modules/bluebird/js/release/promise.js:79:10)

### ISSUE THREAD

**Issue by ladeiko** [OptimalBits/bull#694]

**Comment by ritter** [comment]
Seeing this too using bull@3.2.0

**Comment by ladeiko** [comment]
queue.js contains more than one place where 'return null;' should be placed

**Comment by ritter** [comment]
Possibly duplicate of #518

**Comment by zidsi** [comment]
Duplicate of #518 for sure, but now we talk about bull@3.2.0 

In my case issue pops up when there are delayed jobs in queue when worker processing (queue.process(...)) starts and updateDelaySet is called.

For my case @3.2.0 queue.js there are three places where this script is called and returned promise is not returned/nulled:

```diff
--- a/<html>queue.js (bull@3.2.0)</html>
+++ b/<html><b>queue.js (modified)</b></html>
@@ -260,7 +260,7 @@
         }
       }).catch(function(err){
         queue.emit('error', err);
-      });
+      }).return(null);
     }
   }, queue.settings.guardInterval);
 }
@@ -308,7 +308,7 @@
         if(timestamp){
           _this.updateDelayTimer(timestamp);
         }
-      });
+      }).return(null);
     }).then(function(){
       //
       // Create a guardian timer to revive delayTimer if necessary
@@ -732,7 +732,7 @@
         _this.updateDelayTimer(nextTimestamp);
       }).catch(function(err){
         _this.emit('error', err, 'Error updating the delay timer');
-      });
+      }).return(null);
       _this.delayedTimestamp = Number.MAX_VALUE;
     };
 
```

**Comment by manast** [comment]
closed as duplicate of #518
